### PR TITLE
Upgrade ADTypes to v1

### PR DESCRIPTION
**Pull request by gdalle**
Upgrade ADTypes to v1.0 _for the sparse backend types only_ (nothing yet about the coloring and sparsity detection infrastructure).

Supersedes #294 

**Main changes**

- Bump Julia compat to v1.10, adjust CI
- Remove everything "1.9-specific" in tests
- Replace every `AutoSparseForwardDiff(...)` (construction) with `AutoSparse(AutoForwardDiff(...))`
- Replace every `AutoSparseForwardDiff` (dispatch) with `AutoSparse{<:AutoForwardDiff}` (be careful with type parameters like `{C}`)
- No need to define `AutoSparseEnzyme` anymore
- Use manual trait for dispatch on mode in `sparse_jacobian_cache(ad, args...; kwargs...) = sparse_jacobian_cache_aux(mode(ad), ad, args...; kwargs...)`. Add implem for `ForwardOrReverseMode` which checks if we're dealing with `Diffractor` or `Enzyme` (not hit by tests for some reason :thinking:)
- Replace `_get_colorvec(alg, ad)` with `_get_colorvec(alg, mode(ad))`

**Questions**

- I bumped the version to 2.19 because ColPrac says [dependency compat upgrade is not breaking](https://github.com/SciML/ColPrac?tab=readme-ov-file#changing-dependency-compatibility). However, the ADTypes API did leak into the SparseDiffTools API through the docs, so I'm not sure if this deserves a 3.0?
- In the update process for ADTypes v1.0, we should go from the roots of the dependency tree to the leaves. However, that means downstream integration tests are bound to fail at first. Is that okay?
- This PR does nothing to integrate the coloring and detection infrastructure of SparseDiffTools inside the generic framework introduced by ADTypes v1.0. I think this would be a breaking change, so I suggest doing it separately

### CODE REVIEW

**Comment by gdalle** [comment]
@ChrisRackauckas this is ready for a first pass. Not sure if the missed lines in Codecov are my fault or not.

**Comment by ChrisRackauckas** [comment]
Looks good!

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,10 @@
 name = "SparseDiffTools"
 uuid = "47a9eef4-7e08-11e9-0b38-333d64bd3804"
-authors = ["Pankaj Mishra <pankajmishra1511@gmail.com>", "Chris Rackauckas <contact@chrisrackauckas.com>"]
-version = "2.18.0"
+authors = [
+    "Pankaj Mishra <pankajmishra1511@gmail.com>",
+    "Chris Rackauckas <contact@chrisrackauckas.com>",
+]
+version = "2.19.0"
```
[the remark this answers is not in the document]
Comment (avik-pal) [reply Project.toml:7]: I know ColPrac doesn't require it, but NonlinearSolve is very tightly coupled rn with sparsedifftools..

[the remark this answers is not in the document]
Comment (gdalle) [reply Project.toml:7]: I was wondering whether to include more changes to SparseDiffTools first. Basically I only did the backend part, but the new ADTypes also has infrastructure (supertypes and functions) for sparsity detection + coloring, which SparseDiffTools could extend. On the other hand that would be super super breaking, which is why I didn't do it in the current PR (my priority was getting things to work)

**Pull request merged**